### PR TITLE
Feature added to validate user's attendance

### PR DESCRIPTION
**Pull request by Shiou-Ju**

### CODE REVIEW

File: crawler.py [hunk 1]
```diff
@@ -40,20 +40,31 @@ def login(self):
...
-        print('clockout finish')
-    
-    def mouseMoveAndClick(self, element):
-        action = webdriver.ActionChains(self.driver)
-        action.move_to_element(element).click(element).perform()
-
+        attendance = self.checkUserAttendance()
+        self.mouseMoveAndClick(clockout_div, attendance)
+        
+    def mouseMoveAndClick(self, element, attendance):
```
Comment (mike132129) [inline crawler.py:51]: This function takes moving and clicking action, so the attendance checking should be done in clockin and clockout method.

File: crawler.py [hunk 2]
```diff
@@ -40,20 +40,31 @@ def login(self):
...
+            action = webdriver.ActionChains(self.driver)
+            action.move_to_element(element).click(element).perform()
+            print(f'已打卡{clockType}')
+        else:
+            print(f'無需打卡{clockType}')        
+
+    def checkUserAttendance(self):
+        reminders = self.driver.find_elements_by_id('collapseOne')[0].text     
+        selfName = self.driver.find_elements_by_id('user-dropdown')[0].text.split()[0]
+        if selfName not in reminders:
```
Comment (mike132129) [inline crawler.py:64]: should be 

    if selfName != reminders:

File: crawler.py (hunk 1 above)
Comment (Shiou-Ju) [reply crawler.py:51]: Agree!  I'll revise in the new pull request.

File: crawler.py (hunk 2 above)
Comment (Shiou-Ju) [reply crawler.py:64]: Thanks for reminding!  I didn't notice there might be some cases like what we discussed last time.  Will be revised in the new pull request.

**Pull request closed without merging**